src/meta/src/barrier/recovery.rs

```diff
@@ -326,16 +326,33 @@ impl GlobalBarrierManager {
                         .context
                         .hummock_manager
                         .on_current_version(|version| {
-                            let max_committed_epoch = version.max_committed_epoch_for_meta();
-                            for (table_id, info) in version.state_table_info.info() {
-                                assert_eq!(
-                                    info.committed_epoch, max_committed_epoch,
-                                    "table {} with invisible epoch is not purged",
-                                    table_id
+                            let state_table_info = version.state_table_info.info();
+                            let committed_epoch = state_table_info
+                                .values()
+                                .map(|info| info.committed_epoch)
+                                .next();
+                            let existing_table_ids = info.existing_table_ids();
+                            for table_id in existing_table_ids {
+                                assert!(
+                                    state_table_info.contains_key(&table_id),
+                                    "table id {table_id} not registered to hummock but in recovered job {:?}. hummock table info{:?}",
+                                    info.existing_table_ids().collect_vec(),
+                                    state_table_info
                                 );
                             }
+                            if let Some(committed_epoch) = committed_epoch {
+                                for (table_id, info) in version.state_table_info.info() {
+                                    assert_eq!(
+                                        info.committed_epoch, committed_epoch,
```

Comment:
Just to confirm: the reason why this assertion is valid is because snapshot backfill is non-resumable and tables with fake epochs will be cleaned. In the future with partial ckpt recovery, this assertion will not longer hold.

Reply:
Yes, it's only a sanity check for current implementation.